include/oneapi/dpl/internal/async_impl/async_impl_hetero.h

```diff
@@ -139,15 +135,14 @@ __pattern_transform_reduce_async(_ExecutionPolicy&& __exec, _RandomAccessIterato
         oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _RandomAccessIterator2>();
     auto __buf2 = __keep2(__first2, __first2 + __n);
```

Comment:
This and in the same other places we have very tricky case when we have got host iterators...
In case of the host iterators `__buf2 ` "hides" a temporary sycl::buffer. In case of async call we **have to** propagate the lifetime of one.
Or we should agreed that the async API cannot accept host iterators.

Reply:
Good catch. I'm tempted to not allow direct passing of host iterator for async API. If not, we might think about tagging those temporary iterators. But that might introduce other side effects. For now let's update the documentation and not allow direct passing of host iterators.